### ISSUE TITLE
Question Regarding ScanQA Checkpoint Evaluation

### ISSUE THREAD

**Issue by Yebulabula** [scene-verse/SceneVerse#36]
Dear Author,

Thank you for your excellent contributions to the 3D understanding field.

I am currently evaluating the performance of your provided checkpoints on QA tasks. The SQA3D checkpoints perform as expected on SQA3D; however, when testing the ScanQA checkpoint on the ScanQA validation set, I obtained 0% performance. Could you kindly advise me on this issue?

Best regards,
Ye

<img width="1873" height="694" alt="Image" src="https://github.com/user-attachments/assets/3458ad89-2ce3-4a8c-8d41-c500f323a7c4" />

**Comment by Yebulabula** [comment]
Here is my full log:

[2025-09-21 20:51:08,861] [INFO] [real_accelerator.py:254:get_accelerator] Setting ds_accelerator to cuda (auto detect)
sceneverse_ft_scanqa_b32_ScanQA_ScanNetScanQAOld_multiscan_training_run1
wandb: Currently logged in as: yem to https://api.wandb.ai. Use `wandb login --relogin` to force relogin
wandb: Tracking run with wandb version 0.21.1
wandb: Run data is saved locally in /gpfs/home/ym621/SceneVerse/wandb/run-20250921_205111-i2wy4z4h
wandb: Run `wandb offline` to turn off syncing.
wandb: Syncing run b32_ScanQA_ScanNetScanQAOld_multiscan_training_run1/2025-09-21-20:51:10.736698
name: sceneverse_ft_scanqa
rng_seed: 42
num_gpu: 1
mode: test
note: multiscan_training_run1
naming_keywords:
- dataloader.batchsize
- task
- note
- time
base_dir: results
exp_dir: !!python/object/apply:pathlib.PosixPath
- results
- sceneverse_ft_scanqa_b32_ScanQA_ScanNetScanQAOld_multiscan_training_run1
- 2025-09-21-20:51:10.736698
save_frequency: 10
resume: false
debug:
  flag: false
  hard_debug: false
  debug_size: 20
logger:
  name: wandb
  entity: yem
  run_id: i2wy4z4h
data:
  train:
  - ScanNetScanQAOld
  val:
  - ScanNetScanQAOld
  test:
  - ScanNetScanQAOld
  args:
    max_obj_len: 80
    max_seq_len: 50
    num_points: 1024
    pc_type: pred
    sem_type: '607'
    filter_lang: false
    rot_aug: true
  ScanNetScanQAOld:
    train:
      use_unanswer: true
    val:
      use_unanswer: true
    test:
      use_unanswer: true
      test_file: test_w_obj
  use_voxel: false
  scan_family_base: existing_datasets/ScanNet
  rscan_base: existing_datasets/3RScan
task: ScanQA
data_wrapper:
  train: ScanFamilyDatasetWrapperOld
  val: ScanFamilyDatasetWrapperOld
  test: ScanFamilyDatasetWrapperOld
trainer: DefaultTrainer
ckpt_path: ''
pretrain_ckpt_path: pretrained_weights/scanqa
dataloader:
  batchsize: 32
  num_workers: 2
  balance_dataset: false
  filter_empty_annotations: false
solver:
  gradient_accumulation_steps: 1
  epochs_per_save: 10
  epochs_per_eval: 5
  lr: 0.0001
  grad_norm: 5.0
  epochs: 100
  optim:
    name: AdamW
    args:
      betas:
      - 0.9
      - 0.98
  sched:
    name: warmup_cosine
    args:
      warmup_steps: 5000
eval:
  name: ScanQAEval
  save: false
model:
  name: OpenVocab
  language:
    name: BERTLanguageEncoder
    args:
      weights: bert-base-uncased
      hidden_size: 768
      num_hidden_layers: 4
      num_attention_heads: 12
      type_vocab_size: 2
    lr: 1.0e-05
  vision:
    name: PointOpenVocabEncoder
    args:
      backbone: pointnet++
      hidden_size: 768
      freeze: true
      path: pretrained_weights/pointnetpp-open-bert
      num_attention_heads: 12
      spatial_dim: 5
      num_layers: 4
      dim_loc: 6
      dim_feedforward: 2048
      attn_type: spatial
      pairwise_rel_type: center
      use_matmul_label: false
      lang_type: bert
      lang_path: pretrained_weights/607_text_embeddings
    lr: 0.0001
  grounding:
    name: UnifiedSpatialCrossEncoderV2
    args:
      hidden_size: 768
      num_attention_heads: 12
      num_layers: 4
      dim_feedforward: 2048
      dim_loc: 6
    lr: 0.0001
  inter: before
  heads:
    head_list:
    - qa_head
    qa_head:
      name: QAHeadV1
      args:
        hidden_size: 768
        mlp_size: 256
        glimpse: 1
        flat_out_size: 512
        num_answers: 8864
  loss_type: ListLoss
  loss_list:
  - answer_loss
  - TextObjWithinBatch
  vis_loss_list:
  - answer_loss
  - TextObjWithinBatch
  - test :  ScanNetScanQAOld
Loading ScanNet ScanQA test-set language
total answers is 8864
test_w_obj unanswerable question 0,answerable question 4976
Finish loading ScanNet ScanQA test-set language
Loading ScanNet ScanQA test-set scans
Loading scans: 70 / 70
  0%|                                                                                                                 | 0/70 [00:00<?, ?it/s]/gpfs/home/ym621/SceneVerse/data/datasets/scannet_base.py:58: FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.
  pcd_data = torch.load(os.path.join(self.base_dir, "scan_data",
| 70/70 [00:08<00:00,  7.83it/s]
Finish loading ScanNet ScanQA test-set data
==================================================
Dataset                 Size
ScanNetScanQAOld       4976 (100.0%)
Total                   4976
==================================================
/gpfs/home/ym621/SceneVerse/modules/vision/pcd_openvocab_encoder.py:47: FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.
  self.register_buffer("text_features", torch.load(os.path.join(lang_path, vocab_file_name)))
loaded
/gpfs/home/ym621/SceneVerse/modules/vision/pcd_openvocab_encoder.py:109: FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.
  state_dict = torch.load(ckpt, map_location='cpu')
loaded checkpoint files
Using answer_loss from loss.globals()
Using TextObjWithinBatch from Registry loss
/gpfs/home/ym621/SceneVerse/trainer/build.py:182: FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.
  weights.update(torch.load(model_weight_path, map_location="cpu"))
 | 155/156 [01:29<00:00,  2.16it/s]/gpfs/home/ym621/mambaforge-pypy3/envs/sceneverse/lib/python3.9/site-packages/accelerate/utils/dataclasses.py:301: FutureWarning: The `TPU` of `<enum 'DistributedType'>` is deprecated and will be removed in v1.0.0. Please use the `XLA` instead.
  warnings.warn(
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████| 156/156 [01:29<00:00,  1.75it/s]
wandb: 
wandb: 🚀 View run b32_ScanQA_ScanNetScanQAOld_multiscan_training_run1/2025-09-21-20:51:10.736698 at: https://wandb.ai/yem/sceneverse_ft_scanqa/runs/i2wy4z4h
wandb: Find logs at: wandb/run-20250921_205111-i2wy4z4h/logs
launched
